Add Conda Ecosystem Explained article. (#198)
conda/conda-dot-org: blog/2024-08-14-conda-ecosystem-explained.mdx

authors: [dashagurova, jezdez]
tags: [conda, community, governance]
description: |
  Conda Ecosystem Explained
---

## What is the conda ecosystem?

At its core, the conda ecosystem is a community supporting an open-source, language-agnostic, and multi-platform package management system.

But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components the conda community maintains:

- **CLI Tools for Package Management**: conda command-line tools make it easy to install and manage dependencies in isolated development environments across Linux, Windows, and macOS. Conda works with packages written in any language, e.g. Python, R, C, or Fortran.
- **Package Repository**: most publicly available conda packages are hosted on **[anaconda.org](https://anaconda.org/)**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.
- **Build Tools**: If you want to create and share your own conda packages, the ecosystem provides tools to help you build and distribute them efficiently.

> This system is bolstered by various other supporting projects, all of which you can explore on conda's [GitHub](https://github.com/conda).

Reviewer: ```suggestion
This system is bolstered by various other supporting projects, all of which you can explore on conda's [GitHub](https://github.com/conda) or emerging projects in the [conda-incubator](https://github.com/conda-incubator).
```

And yes, `conda` is free. Always has been, always will be. This commitment to open-source has spurred the development of several compatible alternatives within the conda ecosystem, as well as enhancements to the core `conda` package manager. One notable example is [Mamba](https://mamba.readthedocs.io/), a faster reimplementation in C++. Mamba's speedy solver has been contributed to the core `conda` package manager and is now the [default solver shipped with conda](https://conda.org/blog/2023-11-06-conda-23-10-0-release/). Another emerging tool is [Pixi](https://pixi.sh/), a Rust-based package manager that works seamlessly with conda packages while offering its own unique take on traditional conda workflows.

Furthermore, all of these tools use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

## Conda Package Repository and Channels

> When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.

Reviewer: This sounds a bit like anaconda.org is the only way to host packages. What about the following?

```suggestion
When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages or even run their own package index, we'll focus on the largest and most significant channels within the conda ecosystem.
```

### `conda-forge`

[conda-forge](https://conda-forge.org) is one of the largest and most impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The [governance model](https://conda-forge.org/community/governance/) ensures that the channel remains reliable and accessible to everyone.

### `bioconda`

If your work involves computational biology, [Bioconda](https://bioconda.github.io) is the channel you’ll want to explore. Unlike conda-forge, which covers a wide variety of packages, Bioconda is focused on bioinformatics. It offers over 8,000 packages that cater to subdomains like genomics, proteomics, structural biology, and biomass analysis. Like conda-forge, bioconda is open-source and free to use, maintained by a community of volunteers.

### `defaults`

> The `defaults` channel is maintained by [Anaconda, Inc](https://www.anaconda.com/). It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).

Reviewer: ```suggestion
The `defaults` channel (sometimes referred to as `main`) is maintained by [Anaconda, Inc](https://www.anaconda.com/). It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).
```

        <a href="https://github.com/conda-forge/miniforge">Miniforge</a>
      </td>
      <td>conda & mamba</td>
      <td>`conda-forge`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Micromamba</a>
      </td>
      <td>mamba</td>
      <td>None</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Pixi</a>
      </td>
      <td>pixi</td>
      <td>`conda-forge`</td>
      <td>Emerging</td>
    </tr>
  </tbody>
</table>

Each of these installers offers a different path into the conda ecosystem, allowing you to tailor your setup based on your preferences.

<div className="alert alert--primary" role="alert">
  If you choose Miniconda, which defaults to the `defaults` channel, you can
  easily reconfigure conda after installation to use only free community
  channels like `conda-forge` or `bioconda` by editing your `.condarc file`.

  <a href="https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html">Check the documentation for instructions</a>.

> </div>

Reviewer: ```suggestion
</div>

```

---
title: "Conda Ecosystem Explained"
slug: "2024-08-14-conda-ecosystem-explained"
authors: [dashagurova, jezdez]
tags: [conda, community, governance]
description: |
  Get to know the conda ecosystem—a vibrant community supporting an open-source, language-agnostic, and multi-platform package management system. Discover conda CLI tools for managing packages and environments, and explore key repositories like conda-forge, bioconda and defaults. You will learn how to get started with conda tools, the important things to keep in mind, and how multistakeholder governance ensures the ecosystem remains reliable and accessible to all.
---

> 

Reviewer 2: I would add the same text as in `description` above here as the introduction of the full blog post, and then `<!-- truncate -->`. Otherwise you get the full text in the [blog list](https://deploy-preview-198--conda-dot-org.netlify.app/blog) and is a bit too long for that.

```suggestion

Get to know the conda ecosystem—a vibrant community supporting an open-source, language-agnostic, and multi-platform package management system. Discover conda CLI tools for managing packages and environments, and explore key repositories like conda-forge, bioconda and defaults. You will learn how to get started with conda tools, the important things to keep in mind, and how multistakeholder governance ensures the ecosystem remains reliable and accessible to all.

<!-- truncate -->
```

Author: Good call!

## What is the conda ecosystem?

At its core, the conda ecosystem is a community supporting an open-source, language-agnostic, and multi-platform package management system.

But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components the conda community maintains:

- **CLI Tools for Package Management**: conda command-line tools make it easy to install and manage dependencies in isolated development environments across Linux, Windows, and macOS. Conda works with packages written in any language, e.g. Python, R, C, or Fortran.
- **Package Repository**: most publicly available conda packages are hosted on **[anaconda.org](https://anaconda.org/)**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.
- **Build Tools**: If you want to create and share your own conda packages, the ecosystem provides tools to help you build and distribute them efficiently.

> This system is bolstered by various other supporting projects, all of which you can explore on conda's [GitHub](https://github.com/conda) or emerging projects in the [conda-incubator](https://github.com/conda-incubator).

Reviewer 2: ```suggestion
This system is bolstered by various other supporting projects, all of which you can explore on [conda's GitHub organization](https://github.com/conda) or emerging projects in the [conda-incubator](https://github.com/conda-incubator).
```

## Conda package & environment manager

> The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.

Reviewer 2: ```suggestion
The cornerstone of this entire ecosystem is the [`conda`](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, `conda` is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, `conda` has you covered.
```

Furthermore, all of these tools use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

## Conda Package Repository and Channels

When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages or even run their own package index, we'll focus on the largest and most significant channels within the conda ecosystem.

### `conda-forge`

> [conda-forge](https://conda-forge.org) is one of the largest and most impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The [governance model](https://conda-forge.org/community/governance/) ensures that the channel remains reliable and accessible to everyone.

Reviewer 2: ```suggestion
[conda-forge](https://conda-forge.org) is one of the largest and most impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 26,000 projects and 2,000,000 artifacts, making it a go-to resource for developers across the globe. conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The [governance model](https://conda-forge.org/community/governance/) ensures that the channel remains reliable and accessible to everyone.
```

### `bioconda`

If your work involves computational biology, [Bioconda](https://bioconda.github.io) is the channel you’ll want to explore. Unlike conda-forge, which covers a wide variety of packages, Bioconda is focused on bioinformatics. It offers over 8,000 packages that cater to subdomains like genomics, proteomics, structural biology, and biomass analysis. Like conda-forge, bioconda is open-source and free to use, maintained by a community of volunteers.

### `defaults`

The `defaults` channel (sometimes referred to as `main`) is maintained by [Anaconda, Inc](https://www.anaconda.com/). It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).

> 

Reviewer 2: I would add the mirrors section in this part:

```suggestion

## Other conda packaging servers

...
```

Author: ```suggestion

## Other conda packaging servers

While anaconda.org is the central package repository for a large part of the community, it's by far not the only option to host conda packages:

- A Conda Enhancement Proposal about using Open Container Images (OCI) [for storing conda packages and repodata](https://github.com/conda/ceps/pull/70) is currently discussed, and the [tooling to mirror channels](https://github.com/channel-mirrors/conda-oci-mirror) exists.
- [conda-subchannel](https://github.com/conda-incubator/conda-subchannel) is another community prototype to evolve the way how conda channels can be served through self-hosting or public cloud hosting (e.g. GitHub Pages).
- <a blurb about prefix mirror?>
- Universities and research institutes, as well as many corporate users of conda have been hosting their own packages or mirrors of existing channels within their own infrastructure for various reasons.
```

@jaimergp @wolfv Do you have a good blurb about the conda-forge and/or prefix mirrors? I think it would be excellent to have a broad overview of exists

## Getting Started with conda and conda-compatible tools

There are a few different ways to install and get started with conda (at the time of writing this article):

<table>
  <thead>
    <tr>
      <th>Installer name</th>
      <th>Package manager included</th>
      <th>Automatically sets channel as</th>

>       <th>Status</th>

Reviewer 2: Not sure if this column is necessary.

Author: In what way, unnecessary to highlight that pixi is still emerging?

    <tr>
      <td>
        <a href="https://docs.anaconda.com/miniconda/miniconda-install/">Miniconda</a>
      </td>
      <td>conda</td>
      <td>`defaults`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://github.com/conda-forge/miniforge">Miniforge</a>
      </td>
      <td>conda & mamba</td>
      <td>`conda-forge`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Micromamba</a>
      </td>
      <td>mamba</td>
      <td>None</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Pixi</a>
      </td>
      <td>pixi</td>
      <td>`conda-forge`</td>
      <td>Emerging</td>
    </tr>
  </tbody>
</table>

Each of these installers offers a different path into the conda ecosystem, allowing you to tailor your setup based on your preferences.

<div className="alert alert--primary" role="alert">
  If you choose Miniconda, which defaults to the `defaults` channel, you can
  easily reconfigure conda after installation to use only free community

>   channels like `conda-forge` or `bioconda` by editing your `.condarc file`.

Reviewer 3: ```suggestion
  channels like `conda-forge` or `bioconda` by editing your `.condarc` file.
```

---
title: "Conda Ecosystem Explained"
slug: "2024-08-14-conda-ecosystem-explained"
authors: [dashagurova, jezdez]
tags: [conda, community, governance]

> description: |
>   Conda Ecosystem Explained

Reviewer 4: ```suggestion
description: |
  Get to know the conda ecosystem—a vibrant community supporting an open-source, language-agnostic, and multi-platform package management system. Discover conda CLI tools for managing packages and environments, and explore key repositories like conda-forge, bioconda and defaults. You will learn how to get started with conda tools, the important things to keep in mind, and how multistakeholder governance ensures the ecosystem remains reliable and accessible to all.
```

---

## What is the conda ecosystem?

At its core, the conda ecosystem is a community supporting an open-source, language-agnostic, and multi-platform package management system.

But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components the conda community maintains:

- **CLI Tools for Package Management**: conda command-line tools make it easy to install and manage dependencies in isolated development environments across Linux, Windows, and macOS. Conda works with packages written in any language, e.g. Python, R, C, or Fortran.

> - **Package Repository**: most publicly available conda packages are hosted on **anaconda.org**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.

Reviewer 4: ```suggestion
- **Package Repository**: most publicly available conda packages are hosted on **[anaconda.org](https://anaconda.org/)**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.
```

- **Build Tools**: If you want to create and share your own conda packages, the ecosystem provides tools to help you build and distribute them efficiently.

This system is bolstered by various other supporting projects, all of which you can explore on conda's [GitHub](https://github.com/conda).

## Conda package & environment manager

The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.

`conda` was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.

> And yes, `conda` is free. Always has been, always will be. This commitment to open-source has led to the development of several compatible alternatives within the conda ecosystem, like [Mamba](https://mamba.readthedocs.io/) — a faster reimplementation in C++ \- and [Pixi](https://pixi.sh/), a newer, Rust-based package management tool.

Reviewer 4: ```suggestion
And yes, `conda` is free. Always has been, always will be. This commitment to open-source has spurred the development of several compatible alternatives within the conda ecosystem, as well as enhancements to the core `conda` package manager. One notable example is [Mamba](https://mamba.readthedocs.io/), a faster reimplementation in C++. Mamba's speedy solver has been contributed to the core `conda` package manager and is now the [default solver shipped with conda](https://conda.org/blog/2023-11-06-conda-23-10-0-release/). Another emerging tool is [Pixi](https://pixi.sh/), a Rust-based package manager that works seamlessly with conda packages while offering its own unique take on traditional conda workflows.
```

Furthermore, all of these tools use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

## Conda Package Repository and Channels

> When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. Anaconda.org serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.

Reviewer 4: ```suggestion
When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.
```

> When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. Anaconda.org serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.

Reviewer 5: I think it might be helpful to explicitly say here that use of packages from anaconda.org is not bound by the Anaconda Terms of Service which cover the anaconda/defaults channel.

Author: Thanks for the suggestion, but answering this is out of scope of the article, [Peter has shared a broader plan](https://www.linkedin.com/posts/pzwang_hi-everyone-recently-there-has-been-discussion-activity-7229549723462905856-rQH-/) for what Anaconda Inc is doing to respond to the concerns about the Terms of Services etc. This article here isn't part of that effort and we should stick to elaborating about the conda encosystem.

### `conda-forge`

[conda-forge](https://conda-forge.org) is one of the largest and most impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The governance model ensures that the channel remains reliable and accessible to everyone.

> ### Bioconda

Reviewer 4: ```suggestion
### `bioconda`
```

If your work involves computational biology, [Bioconda](https://bioconda.github.io) is the channel you’ll want to explore. Unlike conda-forge, which covers a wide variety of packages, Bioconda is focused on bioinformatics. It offers over 8,000 packages that cater to subdomains like genomics, proteomics, structural biology, and biomass analysis. Like conda-forge, bioconda is open-source and free to use, maintained by a community of volunteers.

### `defaults`

> The `defaults` channel is maintained by Anaconda, Inc. It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).

Reviewer 4: ```suggestion
The `defaults` channel is maintained by [Anaconda, Inc](https://www.anaconda.com/). It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).
```

## Getting Started with conda and conda-based tools

There are a few different ways to install and get started with conda:

<table>
  <thead>
    <tr>
      <th>Installer</th>
      <th>Package manager included</th>
      <th>Automatically sets channel as:</th>
      <th>Status</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a href="https://docs.anaconda.com/miniconda/miniconda-install/">
          Miniconda
        </a>
      </td>
      <td>conda</td>
      <td>`defaults`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://github.com/conda-forge/miniforge">Miniforge</a>
      </td>

>       <td>conda/mamba</td>

Author: ```suggestion
      <td>conda & mamba</td>
```

      <td>`conda-forge`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">
          Micromamba
        </a>
      </td>
      <td>mamba</td>
      <td>None</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">
          Pixi
        </a>
      </td>
      <td>pixi</td>
      <td>`conda-forge`</td>
      <td>Emerging</td>
    </tr>
  </tbody>
</table>

Each of these installers offers a different path into the conda ecosystem, allowing you to tailor your setup based on your preferences.

<div className="alert alert--primary" role="alert">
  If you choose Miniconda, which defaults to the `defaults` channel, you can
  easily reconfigure conda after installation to use only free community

>   channels like `conda-forge` or `bioonda` by editing your `.condarc file`.{" "}

Reviewer 4: ```suggestion
  channels like `conda-forge` or `bioconda` by editing your `.condarc file`.{" "}
```

>   channels like `conda-forge` or `bioonda` by editing your `.condarc file`.{" "}
>   <a href="https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html">
>     {" "}
>     Check the documentation for instructions
>   </a>
>   .
> </div>

Reviewer 4: ```suggestion
  channels like `conda-forge` or `bioonda` by editing your `.condarc file`.{" "}
  <a href="https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html">
    Check the documentation for instructions.
    </a>
</div>
```

But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components the conda community maintains:

- **CLI Tools for Package Management**: conda command-line tools make it easy to install and manage dependencies in isolated development environments across Linux, Windows, and macOS. Conda works with packages written in any language, e.g. Python, R, C, or Fortran.
- **Package Repository**: most publicly available conda packages are hosted on **[anaconda.org](https://anaconda.org/)**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.
- **Build Tools**: If you want to create and share your own conda packages, the ecosystem provides tools to help you build and distribute them efficiently.

This system is bolstered by various other supporting projects, all of which you can explore on [conda's GitHub organization](https://github.com/conda) or emerging projects in the [conda-incubator](https://github.com/conda-incubator).

## Conda package & environment manager

> The cornerstone of this entire ecosystem is the [`conda`](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, `conda` is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, `conda` has you covered.

Reviewer 6: ```suggestion
The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.
```

The usage of "conda" vs "`conda`" is not consistent; I think the style preferene is to monospace "`conda`" only when referring to a command.

Author: Thank you @beeankha, that's a very good callout!

> `conda` was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.

Reviewer 6: ```suggestion
Conda was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.
```

> And yes, `conda` is free. Always has been, always will be. This commitment to open-source has spurred the development of several compatible alternatives within the conda ecosystem, as well as enhancements to the core `conda` package manager. One notable example is [Mamba](https://mamba.readthedocs.io/), a faster reimplementation in C++. Mamba's speedy solver has been contributed to the core `conda` package manager and is now the [default solver shipped with conda](https://conda.org/blog/2023-11-06-conda-23-10-0-release/). Another emerging tool is [Pixi](https://pixi.sh/), a Rust-based package manager that works seamlessly with conda packages while offering its own unique take on traditional conda workflows.

Reviewer 6: ```suggestion
And yes, conda is free. Always has been, always will be. This commitment to open-source has spurred the development of several compatible alternatives within the conda ecosystem, as well as enhancements to the core conda package manager. One notable example is [Mamba](https://mamba.readthedocs.io/), a faster reimplementation in C++. Mamba's speedy solver has been contributed to the core conda package manager and is now the [default solver shipped with conda](https://conda.org/blog/2023-11-06-conda-23-10-0-release/). Another emerging tool is [Pixi](https://pixi.sh/), a Rust-based package manager that works seamlessly with conda packages while offering its own unique take on traditional conda workflows.
```

Furthermore, all of these tools use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

## Conda Package Repository and Channels

> When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages or even run their own package index, we'll focus on the largest and most significant channels within the conda ecosystem.

Reviewer 6: ```suggestion
When it comes to managing packages with conda or conda-compatible tools, understanding where these packages come from is key. [Anaconda.org](https://anaconda.org/) serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages or even run their own package index, we'll focus on the largest and most significant channels within the conda ecosystem.
```

### `bioconda`

If your work involves computational biology, [Bioconda](https://bioconda.github.io) is the channel you’ll want to explore. Unlike conda-forge, which covers a wide variety of packages, Bioconda is focused on bioinformatics. It offers over 8,000 packages that cater to subdomains like genomics, proteomics, structural biology, and biomass analysis. Like conda-forge, bioconda is open-source and free to use, maintained by a community of volunteers.

### `defaults`

The `defaults` channel (sometimes referred to as `main`) is maintained by [Anaconda, Inc](https://www.anaconda.com/). It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).

## Other conda packaging servers

> While anaconda.org is the central package repository for a large part of the community, it's by far not the only option to host conda packages:

Reviewer 6: ```suggestion
While [anaconda.org](https://anaconda.org/) is the central package repository for a large part of the community, it's by far not the only option to host conda packages:
```

      </td>
      <td>conda & mamba</td>
      <td>`conda-forge`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Micromamba</a>
      </td>
      <td>mamba</td>
      <td>None</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td>
        <a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Pixi</a>
      </td>
      <td>pixi</td>
      <td>`conda-forge`</td>
      <td>Emerging</td>
    </tr>
  </tbody>
</table>

Each of these installers offers a different path into the conda ecosystem, allowing you to tailor your setup based on your preferences.

<div className="alert alert--primary" role="alert">
  If you choose Miniconda, which defaults to the `defaults` channel, you can
  easily reconfigure conda after installation to use only free community
  channels like `conda-forge` or `bioconda` by editing your `.condarc` file.

  <a href="https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html">Check the documentation for instructions</a>.
</div>

## Anaconda's Role in Supporting the conda ecosystem

> Anaconda, Inc. plays a vital role in keeping the conda ecosystem strong and sustainable. Anaconda Inc. contributes through regular donations to NumFOCUS, providing engineering resources, and generously covering hosting fees for anaconda.org. This support is essential for keeping community channels like conda-forge and Bioconda free and accessible.

Reviewer 4: ```suggestion
Anaconda, Inc. (alongside other organizations) plays a vital role in keeping the conda ecosystem strong and sustainable. Anaconda Inc. contributes through regular donations to NumFOCUS, providing engineering resources, and generously covering hosting fees for anaconda.org. This support is essential for keeping community channels like conda-forge and Bioconda free and accessible.
```

> Anaconda, Inc. (alongside other organizations) plays a vital role in keeping the conda ecosystem strong and sustainable. Anaconda Inc. contributes through regular donations to NumFOCUS, providing engineering resources, and generously covering hosting fees for anaconda.org. This support is essential for keeping community channels like conda-forge and Bioconda free and accessible.

Reviewer 6: ```suggestion
Anaconda, Inc. (alongside other organizations) plays a vital role in keeping the conda ecosystem strong and sustainable. Anaconda Inc. contributes through regular donations to NumFOCUS, providing engineering resources, and generously covering hosting fees for [anaconda.org](https://anaconda.org/). This support is essential for keeping community channels like conda-forge and Bioconda free and accessible.
```

---
title: "Conda Ecosystem Explained"
slug: "2024-08-14-conda-ecosystem-explained"
authors: [dashagurova, jezdez]
tags: [conda, community, governance]
description: |

>   Conda Ecosystem Explained

Reviewer 2: This is part of the SEO / search results previews, so we should have a larger blurb here (couple sentences)

---

## What is the conda ecosystem?

At its core, the conda ecosystem is a community supporting an open-source, language-agnostic, and multi-platform package management system.

> But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components conda community maintains:

Reviewer 7: ```suggestion
But what does a "package management ecosystem" really mean? Without diving too deep into every detail, here are the three main components the conda community maintains:
```

- **CLI Tools for Package Management**: conda command-line tools make it easy to install and manage dependencies in isolated development environments across Linux, Windows, and macOS. Conda works with packages written in any language, e.g. Python, R, C, or Fortran.

> - **Package Repository**: most publicly available conda packages are hosted on **anaconda.org**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by community, companies, or individuals.

Reviewer 7: ```suggestion
- **Package Repository**: most publicly available conda packages are hosted on **anaconda.org**, organized into channels. Channels are like remote warehouses where packages are stored and can be pulled from. You can find channels maintained by the community, companies, or individuals.
```

- **Build Tools**: If you want to create and share your own conda packages, the ecosystem provides tools to help you build and distribute them efficiently.

This system is bolstered by various other supporting projects, all of which you can explore on conda's [GitHub](https://github.com/conda).

## Conda package & environment manager

> The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python, R, or a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.

Reviewer 7: ```suggestion
The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python or R or you're a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.
```
Adding clarity on that three item list that's really a two item list.

Reviewer 7: You could leave that second "you're" out, but it's to help with readability since I've added in two "or"s close together.

> Conda was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.

Reviewer 2: ```suggestion
`conda` was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.
```

> And yes, conda is free. Always has been, always will be. This commitment to open-source has led to the development of several compatible alternatives within the conda ecosystem, like [Mamba](https://mamba.readthedocs.io/) — a faster reimplementation in C++ \- and [Pixi](https://pixi.sh/), a newer, Rust-based package management tool.

Reviewer 2: ```suggestion
And yes, `conda` is free. Always has been, always will be. This commitment to open-source has led to the development of several compatible alternatives within the conda ecosystem, like [Mamba](https://mamba.readthedocs.io/) — a faster reimplementation in C++ \- and [Pixi](https://pixi.sh/), a newer, Rust-based package management tool.
```

Reviewer 8: I wonder if we should explain libmamba here. Someone reading this may take away that conda is slow, mamba is fast. However, today the differences are not the solver speed anymore.

Author: I concur, could you suggest an update?

Reviewer 8: You marked as solved but if you still want a suggestion how about?

And yes, `conda` is free. Always has been, always will be. This commitment to open-source has led to the development of several compatible alternatives within the conda ecosystem, like [Mamba](https://mamba.readthedocs.io/) — a re-implementation in C++ \-, libmamba — a faster solver used by conda/mamba, and [Pixi](https://pixi.sh/), a newer, Rust-based package management tool.

> All of which use the same conda standards and formats, that are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

Reviewer 7: ```suggestion
All of which use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.
```
This is technically not a full sentence, but I understand it might be for emphasis. If you wanted to make it a full sentence, replace the first "which" with "these tools". And you could also add a "Furthermore" to the beginning of the sentence for a similar emphasis to what is there already.

Reviewer 7: To be clear, I'm suggesting "Furthermore, all of these tools use the same conda standards and formats, which are..."

## Conda Package Repository and Channels

> When it comes to managing packages with conda or conda-compatible tools, understanding where these packages come from is key. Anaconda.org serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.

Reviewer 2: ```suggestion
When it comes to managing packages with `conda` or conda-compatible tools, understanding where these packages come from is key. Anaconda.org serves as a central hub for conda packages, organized into various channels. While anyone can create their own channel on anaconda.org to host conda packages, we'll focus on the largest and most significant channels within the conda ecosystem.
```

### `conda-forge`

> [conda-forge](https://conda-forge.org) is one of the largest and impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. Conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The governance model ensures that the channel remains reliable and accessible to everyone.

Reviewer 4: ```suggestion
[conda-forge](https://conda-forge.org) is one of the largest and most impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. Conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The governance model ensures that the channel remains reliable and accessible to everyone.
```

> [conda-forge](https://conda-forge.org) is one of the largest and impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. Conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The governance model ensures that the channel remains reliable and accessible to everyone.

Reviewer 2: ```suggestion
[conda-forge](https://conda-forge.org) is one of the largest and impactful channels in the conda ecosystem. Entirely community-driven, this channel hosts over 25,000 packages, making it a go-to resource for developers across the globe. conda-forge is open-source, free to use, and maintained by a group of dedicated volunteers. The governance model ensures that the channel remains reliable and accessible to everyone.
```

### Bioconda

If your work involves computational biology, [Bioconda](https://bioconda.github.io) is the channel you’ll want to explore. Unlike conda-forge, which covers a wide variety of packages, Bioconda is focused on bioinformatics. It offers over 8,000 packages that cater to subdomains like genomics, proteomics, structural biology, and biomass analysis. Like conda-forge, bioconda is open-source and free to use, maintained by a community of volunteers.

> ### `Defaults`

Reviewer 2: ```suggestion
### `defaults`
```

Reviewer 8: It would be nice to get someone from bioconda to comment here but as far as I know bioconda is designed to work with conda-forge and depends on it for some of its packages. See https://bioconda.github.io/#usage

It is a rare case of channel compatibility b/c conda-forge is used to bootstrap bioconda's dependencies.

> The `defaults` channel is maintained by Anaconda, Inc. It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required.

Reviewer 4: ```suggestion
The `defaults` channel is maintained by Anaconda, Inc. It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, [click here](https://www.anaconda.com/pricing/terms-of-service-faqs) to review frequently asked questions about Anaconda’s Terms of Service.
```

Reviewer 2: In the honour of accessibility, we should avoid "here-links". More details at https://www.accessguide.io/guide/descriptive-link-text.

In this case, something like this:

```suggestion
The `defaults` channel is maintained by Anaconda, Inc. It's important to note that this channel comes with [Terms of Service](https://legal.anaconda.com/policies/en/). While largely free for individual users, students, and small companies, there are scenarios where a license might be required. For more information, feel free to review [frequently asked questions about Anaconda’s Terms of Service](https://www.anaconda.com/pricing/terms-of-service-faqs).
```

> ## Getting Started with conda tools

Reviewer 4: ```suggestion
## Getting Started with conda and conda-based tools
```

There are a few different ways to install and get started with conda:

> 1. **Miniconda**: an installer developed by Anaconda, Inc. that installs conda and its dependencies. It is preconfigured to pull packages from the \`defaults\` channel, which has its own Terms of Service (as mentioned in conda channels section). You can always reconfigure conda after installation to use community channels like conda-forge or bioconda \[LINK to docs on how to do it\].

Reviewer 2: ```suggestion
1. **Miniconda**: an installer developed by Anaconda, Inc. that installs `conda` and its dependencies. It is preconfigured to pull packages from the `defaults` channel, which has its own Terms of Service (as mentioned in conda channels section). You can always reconfigure `conda` after installation to use community channels like `conda-forge` or `bioconda` \[LINK to docs on how to do it\].
```

> 1. **Miniconda**: an installer developed by Anaconda, Inc. that installs conda and its dependencies. It is preconfigured to pull packages from the \`defaults\` channel, which has its own Terms of Service (as mentioned in conda channels section). You can always reconfigure conda after installation to use community channels like conda-forge or bioconda \[LINK to docs on how to do it\].

Reviewer 7: Can we link back to an anchor for the conda channels section?

> 2. **Miniforge**: developed by the conda-forge community, this installer comes with both conda and Mamba, and by default, it pulls packages from the conda-forge channel. It's a great option if you prefer a community-driven setup right from the start.

Reviewer 2: ```suggestion
2. **Miniforge**: developed by the conda-forge community, this installer comes with both `conda` and `mamba`, and by default, it pulls packages from the `conda-forge` channel. It's a great option if you prefer a community-driven setup right from the start.
```

Reviewer 8: Maybe, to avoid the default (pre-selected option)  vs defaults (channel) we can phrase this like:

```
this installer comes with both `conda` and `mamba` and it pulls packages from the `conda-forge` channel instead of `defaults`. It's a great option if you prefer a community-driven setup right from the start.
```

> 3. **Micromamba:** developed by the Mamba developers at Quantstack, it’s a statically linked self-contained conda-like package and environment manager, distributed as a standalone executable without a preconfiguration of a default channel.

Reviewer 2: ```suggestion
3. **Micromamba:** developed by the Mamba developers at Quantstack, it’s a statically linked self-contained `conda`-like package and environment manager, distributed as a standalone executable without a preconfiguration of a default channel.
```

> 4. **Pixi**: If you're looking for something a bit different, Pixi is a newer and opinionated conda-compatible package & environment manager written in Rust. Pixi is preconfigured to pull packages from the conda-forge channel.

Reviewer 8: ```
Pixi is preconfigured to pull packages from the conda-forge channel.
```

Is it? pixi has so many entrypoints that this can be confusing. Say, someone is migrating an environment.yml file with pixi but that file has defaults in its configuration, pixi won't remove it. There may be other corner cases but I'm not a heavy user of pixi to comment. I guess that someone with more experience should write on how to config channel in pixi and the possible pitfalls.

> 1. **Miniconda**: an installer developed by Anaconda, Inc. that installs conda and its dependencies. It is preconfigured to pull packages from the \`defaults\` channel, which has its own Terms of Service (as mentioned in conda channels section). You can always reconfigure conda after installation to use community channels like conda-forge or bioconda \[LINK to docs on how to do it\].
> 2. **Miniforge**: developed by the conda-forge community, this installer comes with both conda and Mamba, and by default, it pulls packages from the conda-forge channel. It's a great option if you prefer a community-driven setup right from the start.
> 3. **Micromamba:** developed by the Mamba developers at Quantstack, it’s a statically linked self-contained conda-like package and environment manager, distributed as a standalone executable without a preconfiguration of a default channel.
> 4. **Pixi**: If you're looking for something a bit different, Pixi is a newer and opinionated conda-compatible package & environment manager written in Rust. Pixi is preconfigured to pull packages from the conda-forge channel.

Reviewer 4: ```suggestion
<table>
  <thead>
    <tr>
      <th>Installer</th>
      <th>Package manager included</th>
      <th>Automatically sets channel as:</th>
      <th>Status</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="https://docs.anaconda.com/miniconda/miniconda-install/">Miniconda</a></td>
      <td>conda</td>
      <td>`defaults`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td><a href="https://github.com/conda-forge/miniforge">Miniforge</a></td>
      <td>conda/mamba</td>
      <td>`conda-forge`</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td><a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Micromamba</a></td>
      <td>mamba</td>
      <td>None</td>
      <td>Stable</td>
    </tr>
    <tr>
      <td><a href="https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html">Pixi</a></td>
      <td>pixi</td>
      <td>`conda-forge`</td>
      <td>Emerging</td>
    </tr>
  </tbody>
</table>
```
Changed my mind on the format here for the sake of clarity and readability

Each of these installers offers a different path into the conda ecosystem, allowing you to tailor your setup based on your preferences.

> 

Reviewer 4: ```suggestion
<div className="alert alert--primary" role="alert">
  If you choose Miniconda, which defaults to the `defaults` channel, you can easily reconfigure conda after installation to use only free community channels like `conda-forge` or `bioonda` by editing your `.condarc file`. <a href="https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html"> Check the documentation for instructions</a>.
</div>
```

## Anaconda's Role in Supporting the conda ecosystem

Anaconda, Inc. plays a vital role in keeping the conda ecosystem strong and sustainable. Anaconda Inc. contributes through regular donations to NumFOCUS, providing engineering resources, and generously covering hosting fees for anaconda.org. This support is essential for keeping community channels like conda-forge and Bioconda free and accessible.

> However, it's important to clarify that the conda ecosystem is governed by an independent, diverse community committed to offering free and open source tools and unrestricted package repositories for all users.

Reviewer 4: ```suggestion
However, it's important to clarify that the conda ecosystem is governed by an independent, diverse community committed to offering free and open-source tools and unrestricted package repositories for all users.
See:
- [conda/conda-incubator governance](https://github.com/conda/governance)
- [conda-forge governance](https://conda-forge.org/community/governance/)
- [bioconda core team](https://bioconda.github.io/#core)
```

## Conda package & environment manager

The cornerstone of this entire ecosystem is the [conda](https://github.com/conda/conda) package and environment manager. Originally developed by Anaconda, Inc. and implemented in Python, conda is not limited to Python \- it's a fully capable package manager that handles dependencies across multiple languages and platforms. Whether you're a data scientist working with Python, R, or a machine learning engineer dealing with libraries in C or Fortran, conda has you covered.

Conda was open-source from the start and has grown significantly thanks to support from a broad community. Today, it's [governed by a multi-stakeholder steering council](https://github.com/conda/governance) under the fiscal sponsorship of the [NumFOCUS](https://numfocus.org) non-profit, ensuring it remains a balanced, community-driven project.

And yes, conda is free. Always has been, always will be. This commitment to open-source has led to the development of several compatible alternatives within the conda ecosystem, like [Mamba](https://mamba.readthedocs.io/) — a faster reimplementation in C++ \- and [Pixi](https://pixi.sh/), a newer, Rust-based package management tool.

> All of which use the same conda standards and formats, that are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.

Author: ```suggestion
Furthermore, all of these tools use the same conda standards and formats, which are collaboratively improved through the [Conda Enhancement Proposal (CEP)](https://github.com/conda/ceps/blob/main/cep-1.md) process.
```
